src/Framework/Analyzer/src/DelegateEndpoints/DetectMismatchedParameterOptionality.cs

```diff
@@ -0,0 +1,137 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.AspNetCore.Analyzers.DelegateEndpoints;
+
+public partial class DelegateEndpointAnalyzer : DiagnosticAnalyzer
+{
+    internal const string DetectMismatchedParameterOptionalityRuleId = "ASP0006";
+
+    private static void DetectMismatchedParameterOptionality(
+        in OperationAnalysisContext context,
+        IInvocationOperation invocation,
+        IMethodSymbol methodSymbol)
+    {
+        var value = invocation.Arguments[1].Value;
```

Comment:
I don't think it's required since we know the invocation is of a `MapX` method at this point which has a type signature that requires at least three arguments.

Reply:
I changed my mind about this and added a defensive check to verify the number of arguments. Better safe than sorry.